crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2209,19 +2209,24 @@ impl AuthorityPerEpochStore {
                 .await?
             {
                 ConsensusCertificateResult::SuiTransaction(cert) => {
-                    notifications.push(key);
+                    notifications.push(key.clone());
                     verified_certificates.push(cert);
                 }
                 ConsensusCertificateResult::Defered(deferral_key) => {
-                    // Note: record_consensus_message_processed() must have been called for this
+                    // Note: record_consensus_message_processed() must be called for this
```

Comment:
Could someone remind me how we ensure that deferred tx will eventually be really processed?

Reply:
cc @mystenmark

Reply:
we don't yet, I should have left a TODO. We will need to flush the queue before building the final checkpoint.